Tools/AP_Periph/can.cpp

```diff
@@ -113,14 +113,27 @@ HALSITL::CANIface* AP_Periph_FW::can_iface_periph[HAL_NUM_CAN_IFACES];
  * Variables used for dynamic node ID allocation.
  * RTFM at http://uavcan.org/Specification/6._Application_level_functions/#dynamic-node-id-allocation
  */
-static uint32_t send_next_node_id_allocation_request_at_ms; ///< When the next node ID allocation request should be sent
-static uint8_t node_id_allocation_unique_id_offset;         ///< Depends on the stage of the next request
+static uint32_t send_next_node_id_allocation_request_at_ms[HAL_NUM_CAN_IFACES]; ///< When the next node ID allocation request should be sent
+static uint8_t node_id_allocation_unique_id_offset[HAL_NUM_CAN_IFACES];         ///< Depends on the stage of the next request
```

Comment:
With all these variables with an array of size HAL_NUM_CAN_IFACES... I think it's time to move them into a struct so it's a single variable in an array.

Reply:
I really don't think it makes a difference in terms of readability. And, you would be increasing size by few bytes for struct padding and program as well 😄 

If we were really about abstraction these variables would be member of AP_Periph class and most of the methods here would be member of the class as well. But that's a discussion for another time.

Reply:
i'd prefer the structure list, with this type of loop:
 for (auto &s : interfaces) { }